src/config.rs

```diff
@@ -0,0 +1,124 @@
+use dotenv::dotenv;
+use std::error::Error;
+use std::fmt;
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use toml_edit::{value, DocumentMut, Item};
+
+#[derive(Debug)]
+pub enum ConfigError {
+    FileNotFound(String),
+    ParseError(String),
+    IoError(String),
+}
+
+pub struct Config {
+    pub document: DocumentMut,
+    pub path: PathBuf,
+}
+
+impl fmt::Display for ConfigError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ConfigError::FileNotFound(msg) => write!(f, "File not found: {}", msg),
+            ConfigError::ParseError(msg) => write!(f, "Parse error: {}", msg),
+            ConfigError::IoError(err) => write!(f, "IO error: {}", err),
+        }
+    }
+}
+
+impl Error for ConfigError {}
+
+impl From<io::Error> for ConfigError {
+    fn from(err: io::Error) -> ConfigError {
+        if err.kind() == io::ErrorKind::NotFound {
+            ConfigError::FileNotFound(err.to_string())
+        } else {
+            ConfigError::IoError(err.to_string())
+        }
+    }
+}
+
+impl From<toml_edit::TomlError> for ConfigError {
+    fn from(err: toml_edit::TomlError) -> ConfigError {
+        ConfigError::ParseError(err.to_string())
+    }
+}
+
+impl Config {
+    pub fn new() -> Result<Self, ConfigError> {
+        dotenv().ok();
+
+        // Get the user's home directory
+        let config_dir = dirs::config_dir().unwrap().to_path_buf();
+
+        // Define the config folder
+        let master_mind_folder = config_dir.join("mastermind");
+        match fs::create_dir_all(&master_mind_folder) {
+            Ok(_) => println!("Folder created successfully at {:?}", master_mind_folder),
+            Err(e) => eprintln!("Failed to create folder: {}", e),
+        }
+
+        // Define config file path
+        let config_file = master_mind_folder.join(".config.toml");
```

Comment:
The file name should just be `config.toml`. There's no need to prefix with `.` since it's already inside the config directory.